dataset/batch_image.py

```diff
@@ -1,525 +1,767 @@
-            images = self.get(None, components)
-
-            origin = self._calc_origin(images[0], origin, shape)
-            if shape is None:
-                shape = images.shape[2], images.shape[1]
-            if np.all(np.array(origin) + np.array(shape) > np.array(images.shape[1:3])):
-                shape = images.shape[2] - origin[0], images.shape[1] - origin[1]
-
-            x = slice(origin[0], origin[0] + shape[0])
-            y = slice(origin[1], origin[1] + shape[1])
-            new_images = images[:, y, x].copy()
-
-            setattr(self, components, new_images)
-
-    def _crop_image(self, image, origin, shape):
-        origin = self._calc_origin(image, origin, shape)
-        new_image = image[origin[1]:origin[1] + shape[1], origin[0]:origin[1] + shape[0]].copy()
-        return new_image
+        Resulting shape is obtained as original_shape * factor.
 
-    def _random_crop(self, components='images', shape=None):
-        if shape is not None:
-            images = self.get(None, components)
-            new_images = random_crop_numba(images, shape)
-            setattr(self, components, new_images)
-        return self
+        Parameters
+        -----------
+        components : str
+            component to get an image from
+        p : float
+            probability of applying the transforms
+        image : np.ndarray
+            image to scale
+        factor : float, sequence
+            resulting shape is obtained as original_shape * factor
+            - float - scale all axes with the given factor
+            - sequence (factor_1, factort_2, ...) - scale each axis with the given factor separately
 
-    @action
-    def fliplr(self, components='images'):
-        """ Flip image horizontaly (left / right) """
-        images = self.get(None, components)
-        setattr(self, components, images[:, :, ::-1])
-        return self
+        preserve_shape : bool
+            whether to preserve the shape of the image after scaling
+
+        origin : {'center', 'top_left', 'random'}, sequence
+            Relevant only if `preserve_shape` is True.
+            Position of the scaled image with respect to the original one's shape.
+            - 'center' - place the center of the rescaled image on the center of the original one and crop
+                         the rescaled image accordingly
+            - 'top_left' - place the upper-left corner of the rescaled image on the upper-left of the original one
+                           and crop the rescaled image accordingly
+            - 'random' - place the upper-left corner of the rescaled image on the randomly sampled position
+                         in the original one. Position is sampled uniformly such that there is no need for cropping.
+            - sequence - place the upper-left corner of the rescaled image on the given position in the original one.
+
+        Returns
+        -------
+        np.ndarray : rescaled image
```

Comment:
This docstring is for the`scale` method which returns a batch as it is an action.

Reply:
+
|